ci/daily_tell_slack.yml

```diff
@@ -2,7 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 parameters:
-  success-message: '"$(Agent.JobName) passed: $COMMIT_LINK"'
+  success-message: '\"$(Agent.JobName) passed: $COMMIT_LINK\"'
```

Comment:
I don’t understand the escaping here. iirc (and I did briefly look at the yaml spec to confirm) you only have to escape single quotes within a single-quoted string.

Reply:
They're escaped for Bash, not for YAML. I need the quotes in there to produce valid JSON.

Reply:
Elaborating a little bit, we use this here:
```
MESSAGE="${{ parameters['success-message'] }}"
```
and want that to Azure-resolve to
```
MESSAGE="\"$(Agent.JobName) passed: $COMMIT_LINK\""
```
so we have quotes in there for when we call
```
PAYLOAD="{\"text\":$MESSAGE}"
```
so we produce valid JSON.

Now, you'd think we could have just kept the existing form of
```
PAYLOAD="{\"text\":\"$MESSAGE\n\"}"
```
to have explicit quotes at the point of use. The problem with that is that the alternative, for the perf job, is produced by `jq`, which has to produce a valid JSON string (so it includes all the relevant escaping), and that has embedded quotes. So we need the default case to have embedded quotes too.

Reply:
Makes sense, thanks for the explanation!